src/Workspaces/Core/Portable/LanguageServices/TypeInferenceService/AbstractTypeInferenceService.cs

```diff
@@ -73,6 +73,16 @@ private IEnumerable<ITypeSymbol> Filter(IEnumerable<ITypeSymbol> types, bool fil
                             .ToImmutableReadOnlyListOrEmpty();
             }
 
+            protected IEnumerable<ITypeSymbol> ExpandParamsParameter(IParameterSymbol p)
+            {
+                var result = SpecializedCollections.SingletonEnumerable(p.Type);
+                if (p.IsParams)
+                {
+                    result = result.Concat((p.Type as IArrayTypeSymbol).ElementType);
```

Comment:
Also, what about error handling cases where you do params with a non-array? What does the compiler give us there?